main/src/ui/add_conversation/list_row.vala

```diff
@@ -8,27 +8,55 @@ namespace Dino.Ui {
 
 public class ListRow : Widget {
 
-    public Grid outer_grid;
+    public Box outer_box;
     public AvatarPicture picture;
     public Label name_label;
+    public Image status_dot;
     public Label via_label;
 
+    public string? status_str;
     public Jid? jid;
     public Account? account;
 
     construct {
         Builder builder = new Builder.from_resource("/im/dino/Dino/add_conversation/list_row.ui");
-        outer_grid = (Grid) builder.get_object("outer_grid");
+        outer_box = (Box) builder.get_object("outer_box");
         picture = (AvatarPicture) builder.get_object("picture");
         name_label = (Label) builder.get_object("name_label");
+        status_dot = (Image) builder.get_object("status_dot");
         via_label = (Label) builder.get_object("via_label");
 
+
         this.layout_manager = new BinLayout();
-        outer_grid.set_parent(this);
+        outer_box.set_parent(this);
     }
 
     public ListRow() {}
 
+    private void set_status_dot(StreamInteractor stream_interactor, Jid jid, Account account){
+        Gee.List<Jid>? full_jids = stream_interactor.get_module(PresenceManager.IDENTITY).get_full_jids(jid, account);
+        string presence_str = null;
+        if (full_jids != null){
+            int devices = 0;
+            string newline_if_more_devices;
+            for (int i = 0; i < full_jids.size; i++) {
+                Jid full_jid = full_jids[i];
+                presence_str = stream_interactor.get_module(PresenceManager.IDENTITY).get_last_show(full_jid, account);
+                if(presence_str != null) devices += 1;
+                newline_if_more_devices = devices > 1 ? "\r\n" : "";
+                switch(presence_str){
+                        case "dnd": this.status_dot.set_from_icon_name("dino-status-dnd"); i = full_jids.size; break; // dnd detected = marked as dnd for all devices
+                        case "online": this.status_dot.set_from_icon_name("dino-status-online"); i = full_jids.size; break;
+                        case "away": this.status_dot.set_from_icon_name("dino-status-away"); break;
+                        case "xa": this.status_dot.set_from_icon_name("dino-status-away"); break;
+                        case "chat": this.status_dot.set_from_icon_name("dino-status-chat"); break;
```

Comment:
The way this is written it doesn't allow prioritization of statuses, as the order of resources in the full_jids list affects the outcome. For example, if the user has one online and one dnd device, we'd want the status to be dnd, however if the list is [online, dnd], this routine would return online.
One solution would be to first collect all statuses in a list and then check one after another:
- If the list contains "dnd", then return dnd
- If it contains "chat", then return chat
- If it contains online, return online

And so on.

Reply:
@fiaxh Oh yeah thinking about it it's more suitable for presence than an ole' switch statement, will do.